samples/README.md

```diff
@@ -1,27 +1,62 @@
-The sample pipelines give you a quick start to building and deploying machine learning pipelines with Kubeflow.
+The sample pipelines give you a quick start to build and deploy machine learning pipelines with Kubeflow Pipeline.
 * Follow the guide to [deploy the Kubeflow pipelines service](https://www.kubeflow.org/docs/guides/pipelines/deploy-pipelines-service/).
 * Build and deploy your pipeline [using the provided samples](https://www.kubeflow.org/docs/guides/pipelines/pipelines-samples/).
 
-
-
-
-This page tells you how to use the _basic_ sample pipelines contained in the repo.
+# Sample Structure
+The samples are organized into the core set and the contrib set. 
+
+**Core samples** demonstrates the full KFP functionalities and covered by the sample test infra. 
+The current status is not all samples are covered by the tests but they will be all covered in the near future.
+A selected set of these core samples will also be preloaded to the KFP during deployment. 
+The core samples will also include intermediate samples that are 
+more complex than basic samples such as flip coins but simpler than TFX samples. 
+It serves to demonstrate a set of the outstanding features and offers users the next level KFP experience.
+
+**Contrib samples** are not tested by KFP and could potentially be moved to
+the core samples if the samples are of good qualty and tests are covered and it demonstrates certain KFP functionality. 
+Another reason to put some samples in this directory is that some samples require certain 
+platform support that is hard to support in our test infra.
+
+In the Core directory, each sample will be in a separate directory. 
+In the Contrib directory, there is an intermediate directory for each contributor, 
+e.g. ibm and arena, within which each sample is in a separate directory. 
+An example of the resulting structure is as follows: 
+```
+pipelines/samples/
+Core/
+	DSL Static Type Checking/
```

Comment:
shall we adopt a more pythonic naming convention here? like 'dsl_static_type_checking'?
I'm a bit of worrying that adding spaces in dir name might be error-prone.

Reply:
Updated.

Reply:
I copied these from the PR description: https://github.com/kubeflow/pipelines/pull/1710